utxo/src/utxo_impl/utxo_storage/in_memory.rs

```diff
@@ -0,0 +1,71 @@
+use std::collections::{BTreeMap, HashMap};
+
+use super::{UtxosStorageRead, UtxosStorageWrite};
+use crate::{BlockUndo, Utxo};
+use chainstate_types::storage_result::Error;
+use common::{
+    chain::{Block, GenBlock, OutPoint},
+    primitives::Id,
+};
+
+#[derive(Clone)]
+pub struct UtxosDBInMemoryImpl {
+    store: BTreeMap<OutPoint, Utxo>,
+    undo_store: BTreeMap<Id<Block>, BlockUndo>,
+    best_block_id: Option<Id<GenBlock>>,
```

Comment:
Why is this an `Option`? Would it be appropriate to just initialise best block ID upon creation and be sure one exists from that point on? Also how is this different from best block ID in regular chainstate?

Reply:
I'll give it a shot to remove the Option.

Btw, you asked the question of why it's not the same best block before. The answer is that the "utxo set" representation is isolated and separate from the rest of the database. The best block id, as part of the UtxosView is a representation of "what utxo set is represented here". Now, when we combine that with the rest of the node, then we can say that the whole BlockchainStorage is represented by the same best block. So to make them one, this should be done at the BlockchainStorage implementation level, but then if we do that, then this could get the utxo set wrapper (UtxosDB) to replace the BlockchainStorage value that's given by the block connection process, which is called when activating best chain. So, we either merge everything utxo related to the BlockchainStorage and get rid of this interface (which will kill the beautiful hierarchy created with UtxosCache), or we live with having two best blocks because both represent the same thing in two different contexts. Maybe there's a better solution I'm not seeing. I don't know.